### PR TITLE
config: Add a flag to control the /metrics and /status endpoints

### PR DESCRIPTION
**Pull request by freesteph**
### Component/Part
Application

### Description
It can be a security concern in some environments to expose system capabilities even though they don't expose any PII. Add a couple flags to control whether the `/metrics` and `/status` (and anything in the StatusRouter) are exposed.

### Steps

- [X] Added implementation
- [ ] Added / updated tests
- [X] Added / updated documentation
- [X] Added changelog snippet
- [X] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### Related Issue(s)
https://github.com/hedgedoc/hedgedoc/issues/2574

### CODE REVIEW

File: docs/content/configuration.md [hunk 1]
```diff
@@ -76,24 +76,27 @@ these are rarely used for various reasons.
...
+| `cookiePolicy`                | `CMD_COOKIE_POLICY`                | **`lax`**, `strict` or `none`                                                             | Set a SameSite policy whether cookies are send from cross-origin. Be careful: setting a SameSite value of none without https breaks the editor.                           |
+| `observability.exposeMetrics  | `CMD_OBSERVABILITY_EXPOSE_METRICS` | **true** or `false`                                                                       | Controls whether the /metrics endpoint is exposed.                                                                                                                        |
+| `observability.exposeMetrics  | `CMD_OBSERVABILITY_EXPOSE_STATUS`  | **true** or `false`                                                                       | Controls whether the /status routes are exposed.                                                                                                                          |
```
Comment (DerMolly) [inline docs/content/configuration.md:98]:
```suggestion
| `observability.exposeStatus  | `CMD_OBSERVABILITY_EXPOSE_STATUS`  | **true** or `false`                                                                       | Controls whether the /status routes are exposed.                                                                                                                          |
```

**Review by DerMolly — CHANGES_REQUESTED** [review]

**Review by mrdrogdrog — CHANGES_REQUESTED** [review]
Will approve as soon as docs are fixed

**Comment by freesteph** [comment]
All done I believe. Thanks for looking into this!

**Review by ErikMichelson — CHANGES_REQUESTED** [review]
Two points:
- The config option is quite long with the term “observability” inside. In my opinion, `CMD_EXPOSE_METRICS` would be good enough.
- Having the status endpoint disabled, will break the [docker image's health check](https://github.com/hedgedoc/container/blob/master/resources/healthcheck.mjs). Therefore, I suggest that queries from localhost will always be allowed, no matter the config value.

**Review by DerMolly — APPROVED** [review]

**Comment by mrdrogdrog** [comment]
Regarding the docker healthcheck we could use a new route that just returns "ok".

**Comment by freesteph** [comment]
> Two points:
> 
>     * The config option is quite long with the term “observability” inside. In my opinion, `CMD_EXPOSE_METRICS` would be good enough. This is not very important to me, but I wanted to note it anyway.
> 
>     * Having the status endpoint disabled, will break the [docker image's health check](https://github.com/hedgedoc/container/blob/master/resources/healthcheck.mjs). Therefore, I suggest that queries from localhost will always be allowed, no matter the config value.

Thanks for the feedback. I think it makes sense to group these two options into an `observability` scope, it allows grouping them which is nice. Up to you though.

I've added a commit to keep `/status` returning 200, let me know if that's good enough.

**Review by ErikMichelson — APPROVED** [review]

**Review by mrdrogdrog — APPROVED** [review]

**Review by InnayTool — APPROVED** [review]

**Pull request merged**